.github/workflows/ci.yml

```diff
@@ -1,51 +1,54 @@
 name: CI
 
 on:
-  # Triggers the workflow on push or pull request events but only for the master branch
+  workflow_dispatch:
   push:
-    branches: [ master, develop]
+    branches:
+      - master
+      - develop
   pull_request:
-    branches: [ master, develop]
-
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
 
 jobs:
-  build:
+  build_application:
+    name: Build application using Ledger's reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      upload_app_binaries_artifact: compiled_app_binaries
+      skip_stax: true
+  
+  scan_build:
+    name: Perform static analysis using Ledger's reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/_check_clang_static_analyzer.yml@v1
+  
+  ragger_tests:
+    name: Functional tests with Ragger
+    needs: build_application
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - device: nanos
+          - device: nanox
     runs-on: ubuntu-latest
-
-    container:
-      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
-
     steps:
-      - uses: actions/checkout@v2
-
-      - name: Build app
-        run: |
-          make clean
-          make DEBUG=1
-      - name: Upload app binary
-        uses: actions/upload-artifact@v2
+      - name: Clone
+        uses: actions/checkout@v3
+      - name: Download app binaries
+        uses: actions/download-artifact@v3
         with:
-          name: app-debug
-          path: bin
-
-  scan-build:
-    name: Clang Static Analyzer
-    runs-on: ubuntu-latest
-
-    container:
-      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
-
-    steps:
-      - uses: actions/checkout@v2
-
-      - name: Build with Clang Static Analyzer
+          name: compiled_app_binaries
+          path: build
+      - name: Install tests dependencies
         run: |
-          make clean
-          scan-build --use-cc=clang -analyze-headers -enable-checker security -enable-checker unix -enable-checker valist -o scan-build --status-bugs make default
-      - uses: actions/upload-artifact@v2
-        if: failure()
-        with:
-          name: scan-build
-          path: scan-build
+          sudo apt-get update && sudo apt-get install -y qemu-user-static tesseract-ocr libtesseract-dev
+          pip install --upgrade pip
+          pip install --extra-index-url https://test.pypi.org/simple/ -r requirements.txt
+      - name: Build proto
+        run: |
+          cd examples
+          ./buildproto.sh  
```

Comment:
It's weird that it's in `examples` folder?
I'm also wondering, does it works if calls with `examples/buildproto.sh`?

Reply:
@xchapron-ledger , seriously, this part is absolutely not documented and having to tinker it to make it work was already a pain in the ass so... I don't know why it's this way, but now it works. If you don't mind, we can keep it this way.

Reply:
yep, that a job for another day  :)